### ISSUE TITLE
JavaScript vm broken (bad Proxy implementation?)

### ISSUE THREAD

**Issue by freehuntx** [stackblitz/webcontainer-core#309]
### Describe the bug

When defining properties on a object returned by the vm, they behave differently than in NodeJS.  
This breaks for example the testing framework Jest. (As you can see here: https://github.com/stackblitz/webcontainer-core/issues/293)

Example code:
```
const { runInContext, createContext } = require('vm');

const context = createContext();
const vmGlobal = runInContext('this', context);

Object.defineProperties(vmGlobal, {
  works: {
    configurable: true,
    value: 'This will work'
  },
  worksNot: {
    configurable: false,
    value: 'This will not work (in webcontainer)'
  }
});

console.log(vmGlobal.works); // success
console.log(vmGlobal.worksNot); // error
Object.keys(vmGlobal); // error
```

### Link to the blitz that caused the error

https://stackblitz.com/edit/node-7mnp6x

### Steps to reproduce

1. Open the linked stackblitz
2. run `node index` in the console
3. See the error

### Expected behavior

Console should print:
```
This will work
This will not work (in webcontainer)
```

### Screenshots

![image](https://user-images.githubusercontent.com/12412020/130804373-0f387afe-0fd6-457f-81d6-1922d30b0776.png)

### Platform

```
Browser name  = Chrome 
Full version  = 92.0.4515.159
Major version = 92
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.159 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 96025376,
  "usedJSHeapSize": 90858908,
  "jsHeapSizeLimit": 4294705152
}
Hash = 8ffc536d8181e8b30c228dec423d18afae547c40
```

### Additional context

I created this bug ticket as a result of this: https://github.com/stackblitz/webcontainer-core/issues/293

**Comment by d3lm** [comment]
Thanks for reporting this issue. It seems to be a bug.

**Comment by d3lm** [comment]
Also, this seems to work just fine for me locally:
```
console.log(runInContext('global', createContext()));
```

**Comment by d3lm** [comment]
Good news, this has been fixed and it's now available. Here's an [example](https://stackblitz.com/edit/node-khi4at).

**Comment by freehuntx** [comment]
You guys are awesome! It works, thanks!

**Comment by d3lm** [comment]
@freehuntx Just a quick update that `vm.runInContext('global', vm.createContext())` won't behave the same as on local. I had to change this again because the way I fixed it does not work in all cases and `typeof global` was also throwing a reference error. Unfortunately, this is a limitation of WebContainer's `vm` module. The closest I can get is to return `undefined` if you try to evaluate `global` and the context doesn't define it. It's a bummer but sadly a hard limitation. But I'd say this is an edge case, and the rest still works as expected.
